client/python/gradio_client/client.py

```diff
@@ -0,0 +1,320 @@
+"""The main Client class for the Python client."""
+from __future__ import annotations
+
+import concurrent.futures
+import json
+import re
+import uuid
+from concurrent.futures import Future
+from typing import Callable, Dict, List
+
+import requests
+import websockets
+from gradio_client import serializing, utils
+from packaging import version
+
+
+class Client:
+    def __init__(
+        self,
+        space: str | None = None,
+        src: str | None = None,
+        access_token: str | None = None,
+        max_workers: int = 40,
+    ):
+        self.access_token = access_token
+        self.headers = (
+            {"Authorization": "Bearer {access_token}"} if access_token else {}
+        )
+
+        if space is None and src is None:
+            raise ValueError("Either `space` or `src` must be provided")
+        elif space and src:
+            raise ValueError("Only one of `space` or `src` must be provided")
+        self.space = space
```

Comment:
It seems `self.space` is never used except in `__init__.py` to compute the src url.
I would rather not set it and pass an arg to `_space_name_to_src`.

Reply:
Done